pkg/rke2/psp/psp.go

```diff
@@ -0,0 +1,443 @@
+							return err
+						}
+					}
+				}
+				ns.SetAnnotations(map[string]string{namespaceAnnotationGlobalRestricted: cisAnnotationValue})
+			}
+
+			if err := setSystemUnrestricted(sCtx, cs, ns); err != nil {
+				return err
+			}
+
+			if _, ok := ns.Annotations[namespaceAnnotationGlobalUnrestricted]; !ok {
+				if _, err := cs.PolicyV1beta1().PodSecurityPolicies().Get(sCtx, globalRestrictedPSPName, metav1.GetOptions{}); err != nil {
+					if apierrors.IsNotFound(err) {
+						tmpl := fmt.Sprintf(globalRestrictedPSPTemplate, globalRestrictedPSPName)
+						if err := deployPodSecurityPolicyFromYaml(sCtx, cs, tmpl); err != nil {
+							return err
+						}
+					}
+				}
+
+				// check if role exists and if so, delete it
+				role, err := cs.RbacV1().ClusterRoles().Get(sCtx, globalRestrictedRoleName, metav1.GetOptions{})
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						logrus.Info(err)
+					}
+				} else {
+					if err := cs.RbacV1().ClusterRoles().Delete(sCtx, role.Name, metav1.DeleteOptions{}); err != nil {
+						return err
+					}
+				}
+
+				// check if role binding exists and if so, delete it
+				roleBinding, err := cs.RbacV1().ClusterRoleBindings().Get(sCtx, globalUnrestrictedRoleName, metav1.GetOptions{})
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						logrus.Info(err)
+					} else {
+						if err := cs.RbacV1().ClusterRoleBindings().Delete(sCtx, roleBinding.Name, metav1.DeleteOptions{}); err != nil {
+							return err
+						}
+					}
+				}
+
+				ns.SetAnnotations(map[string]string{namespaceAnnotationGlobalUnrestricted: cisAnnotationValue})
+			}
+		}
+
+		// apply node cluster role binding regardless of whether we're in CIS mode or not
+		if _, err := cs.RbacV1().ClusterRoleBindings().Get(sCtx, globalUnrestrictedRoleName, metav1.GetOptions{}); err != nil {
+			if apierrors.IsAlreadyExists(err) {
+				logrus.Info(err)
+			} else {
+				tmpl := fmt.Sprintf(nodeClusterRoleBindingTemplate, globalUnrestrictedPSPName)
+				if err := deployClusterRoleBindingFromYaml(sCtx, cs, tmpl); err != nil {
+					return err
+				}
+			}
+		}
+		return nil
+	}
+}
+
+type deployFn func(context.Context, *kubernetes.Clientset, interface{}) error
+
+// newClient create a new Kubernetes client from configuration.
+func newClient(kubeConfigPath string, k8sWrapTransport transport.WrapperFunc) (*kubernetes.Clientset, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	if err != nil {
+		return nil, err
+	}
+	if k8sWrapTransport != nil {
+		config.WrapTransport = k8sWrapTransport
+	}
+	config.Timeout = time.Second * defaultTimeout
+	return kubernetes.NewForConfig(config)
+}
+
+func decodeYamlResource(data interface{}, yaml string) error {
+	decoder := yamlutil.NewYAMLToJSONDecoder(bytes.NewReader([]byte(yaml)))
+	return decoder.Decode(data)
+}
+
+func retryTo(runFunc deployFn, ctx context.Context, cs *kubernetes.Clientset, resource interface{}, retries, wait int) error {
```

Comment:
Can we use k8s.io/client-go/util/retry instead?

Reply:
Sure. This was just lifted from RKE.

Reply:
nice-to-have. can be left for another day if time is an issue

Reply:
Issue created.